### PR TITLE
Indicate which elements are out of order. Rather than indicating only…

### PR DESCRIPTION
**Pull request by stromkos**
… missing element.

It would be better to point the user to a documentation page indicating proper XML Schema for additional info. See #6338.

### CODE REVIEW

**Comment by stromkos** [comment]
This update gives feedback to the user, when an invalid XML element is encountered; and it reports the
expected element , as well as, the encountered element.
This will inform the user an out of order element was encountered and which element was expected.

**Comment by nkast** [comment]
There are other reasons this error can occur. The element could be renamed, or missing.
Let's not propose a solution which is in 1/3 of the cases.

**Comment by Jjagg** [comment]
> Let's not propose a solution which is in 1/3 of the cases.

Agreed. Though it might be helpful to note that the elements are ordered without suggesting it's what's wrong.

**Comment by stromkos** [comment]
I want to give the user more information more information to correct the error. If it is renamed or deleted this PR will notify the user of the discrepancy. I have reworded the error.  While this may not cover all of the scenarios at which this error may occur, at least it assists the user by notifying them of the expected element.  I was at a loss when I received this message and would like to make sure future users do not have the same issue.

**Comment by stromkos** [comment]
@nkast In your view 2/3s of the cases are currently excluded from the existing wording *without explanation*.  I have posted an update that includes my updates and the existing case.  I have purposefully excluded the rename case as it has zero application in this discussion(no one would expect a malformed XML file to process), although if it did happen, the message would indicate it as a found/expected pair, indicating to the user the corrective action(n a major improvement from the current situation).

**Comment by nkast** [comment]
>> @nkast In your view 2/3s of the cases are currently excluded from the existing wording without explanation. 

On the contrary, I would prefer a simple message with only the facts about the failure. Not an extended list of all possible causes/fixes. 
The problem I saw with the first wording was that it might be confusing in some cases.  Anyways, the new wording is not bad.

**Comment by nkast** [comment]
Is it possible to get Line numbers from the XmlReader (IXmlLineInfo interface)?

**Comment by Jjagg** [comment]
IIRC you need to explicitly cast to IXmlLineInfo as it's implemented as an explicit interface implementation.

File: MonoGame.Framework.Content.Pipeline/Serialization/Intermediate/ReflectiveSerializer.cs [hunk 1]
```diff
@@ -174,7 +174,7 @@ protected internal override object Deserialize(IntermediateReader input, Content
                             continue;
                         
                         // We failed to find a required element.
-                        throw new InvalidContentException(string.Format("The Xml element `{0}` is required!", info.Attribute.ElementName));
+                        throw new InvalidContentException(string.Format("The Xml element `{0}` is required, But found '{1}' instead, try swappping elements as they may be out of order, or required elements may be missing altogether", info.Attribute.ElementName, input.Xml.Name));
```
Comment (tomspilman) [inline MonoGame.Framework.Content.Pipeline/Serialization/Intermediate/ReflectiveSerializer.cs]: Some grammar fixes:

```
The Xml element `{0}` is required, but element `{1}` was found.  Try changing the element order or adding missing elements.
```

Comment (stromkos) [reply MonoGame.Framework.Content.Pipeline/Serialization/Intermediate/ReflectiveSerializer.cs]: Do you want the line number as well?
```
The Xml element `{0}` is required, but element `{1}` was found at line {2}. Try changing the element order or adding missing elements.", info.Attribute.ElementName, input.Xml.Name,((IXmlLineInfo)input.Xml).LineNumber);
```
Or shorter wording with position:
```
The Xml element `{0}` is required, but element `{1}` was found at {2},{3}. Try changing the element order or adding missing elements.", info.Attribute.ElementName, input.Xml.Name,((IXmlLineInfo)input.Xml).LineNumber, ((IXmlLineInfo)input.Xml).LinePosition);
```

**Comment by stromkos** [comment]
Does anyone object to the addition of the Xml Namespace(it is tied to at least functions of the class,which justifies the using statement)?(if not, I will fully qualify the IXmlLineInfo as appropriate)
If everyone approves of the new wording @tomspilman, @nkast, @Jjagg, @cra0zy .  I am always open to suggestions.  
Thank You.
